cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -438,6 +438,103 @@ func isNodeSelectorUpdated(old *v1.NodeSelector, new *v1.NodeSelector) bool {
 	return !reflect.DeepEqual(old.NodeSelectorTerms, new.NodeSelectorTerms)
 }
 
+// isProtocolConfigUpdated checks if protocol is updated
+func isProtocolConfigUpdated(old *v1alpha1.ProtocolConfig, new *v1alpha1.ProtocolConfig) bool {
+	return !reflect.DeepEqual(old, new)
+}
+
+// updateProtocolInConfigMap updates the protocol in the deviceProfile in configmap
+func (dc *DownstreamController) updateProtocolInConfigMap(device *v1alpha1.Device) {
+	if len(device.Spec.NodeSelector.NodeSelectorTerms) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions) != 0 && len(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values) != 0 {
+		configMap, ok := dc.configMapManager.ConfigMap.Load(device.Spec.NodeSelector.NodeSelectorTerms[0].MatchExpressions[0].Values[0])
+		if ok {
+			nodeConfigMap, ok := configMap.(*v1.ConfigMap)
+			if !ok {
+				log.LOGGER.Errorf("Failed to assert to configmap")
+				return
+			}
+			dp, ok := nodeConfigMap.Data[DeviceProfileJSON]
+			if !ok || dp == "{}" {
+				// This case should never be hit as we delete empty configmaps
+				log.LOGGER.Errorf("Failed to get deviceProfile from configmap data or deviceProfile is empty")
+				return
+			}
+
+			deviceProfile := &types.DeviceProfile{}
+			err := json.Unmarshal([]byte(dp), deviceProfile)
```

Comment:
I like the format of

```
if err := xxx; err != nil {
  xxx
}
```

to save one line :)

Reply:
Done, Thanks